activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -20,14 +20,313 @@
+            ['body_type', 'age', 'fuel_type']].astype(str).agg('_'.join, axis=1)
+        locals_dict.update({'vehicle_type_data': vehicle_type_data})
+
+    # - Preparing alternatives
+    # create alts on-the-fly as cartesian product of categorical values
+    if alts_cats_dict:
+        # do not include fuel types as alternatives if probability file is supplied
+        if probs_spec_file:
+            del alts_cats_dict['fuel_type']
+        alts_wide, alts_long = get_combinatorial_vehicle_alternatives(alts_cats_dict, model_settings)
+
+        # merge vehicle type data to alternatives if data is provided
+        if vehicle_type_data_file and (probs_spec_file is None):
+
+            alts_wide = pd.merge(
+                alts_wide, vehicle_type_data,
+                how='left',
+                on=['body_type', 'fuel_type', 'age'],
+                indicator=True)
+
+            # checking to make sure all alternatives have data
+            missing_alts = alts_wide.loc[
+                alts_wide._merge == 'left_only', ['body_type', 'fuel_type', 'age']]
+
+            assert len(missing_alts) == 0, \
+                f"missing vehicle data for alternatives:\n {missing_alts}"
+            alts_wide.drop(columns='_merge', inplace=True)
+
+        # converting age to integer to allow interactions in utilities
+        alts_wide['age'] = alts_wide['age'].astype(int)
+
+    # - preparing choosers for iterating
+    vehicles_merged = vehicles_merged.to_frame()
+    vehicles_merged['already_owned_veh'] = ''
+    logger.info("Running %s with %d vehicles", trace_label, len(vehicles_merged))
+    all_choosers = []
+    all_choices = []
+
+    # - Selecting each vehicle in the household sequentially
+    # This is necessary to determine and use utility terms that include other
+    #   household vehicles
+    for veh_num in range(1, vehicles_merged.vehicle_num.max()+1):
+        # - preprocessor
+        # running preprocessor on entire vehicle table to enumerate vehicle types
+        # already owned by the household
+        choosers = vehicles_merged
+        preprocessor_settings = model_settings.get('preprocessor', None)
+        if preprocessor_settings:
+            expressions.assign_columns(
+                df=choosers,
+                model_settings=preprocessor_settings,
+                locals_dict=locals_dict,
+                trace_label=trace_label)
+
+        # only make choices for vehicles that have not been selected yet
+        choosers = choosers[choosers['vehicle_num'] == veh_num]
+        logger.info("Running %s for vehicle number %s with %d vehicles", trace_label, veh_num, len(choosers))
+
+        # if there were so many alts that they had to be created programmatically,
+        # by combining categorical variables, then the utility expressions should make
+        # use of interaction terms to accommodate alt-specific coefficients and constants
+        if alts_cats_dict:
```

Comment:
The decision to use interaction terms seems like an assumption that we don't need to (and shouldn't) make implicit in the code - rather the user should declare it explicitly in the settings.

Reply:
Added "SIMULATION_TYPE" option to the config yaml so the user has to specify explicitly.  Comments in the config were also included.